src/ruleset.rs

```diff
@@ -558,20 +572,69 @@ impl RulesetCreated {
     }
 }
 
-impl Drop for RulesetCreated {
-    fn drop(&mut self) {
-        if self.fd >= 0 {
-            unsafe { close(self.fd) };
+macro_rules! define_rulset_restrict_object {
+    ($t:ident) => {
+        #[cfg_attr(test, derive(Debug))]
+        pub struct $t {
+            fd: RawFd,
+            no_new_privs: bool,
+            #[allow(unused)]
+            pub(crate) requested_handled_fs: BitFlags<AccessFs>,
+            compat: Compatibility,
         }
-    }
-}
 
-impl AsMut<RulesetCreated> for RulesetCreated {
-    fn as_mut(&mut self) -> &mut RulesetCreated {
-        self
-    }
+        impl $t {
+            #[allow(unused)]
+            fn new(ruleset: Ruleset, fd: RawFd) -> Self {
+                $t {
+                    fd,
+                    no_new_privs: true,
```

Comment:
I'll merge #12 in a few day and this will change some code (e.g. no_new_privs), but I want to update the CI tests before (cf. #41). You can already rebase on #12.